src/wallet.js

```diff
@@ -441,7 +441,7 @@ const wallet = {
     // Updating address because the last one was used
     if (this.hasNewAddress()) {
       this.getNextAddress();
-    } else {
+    } else if (this.canGenerateNewAddress()) {
       this.generateNewAddress();
     }
```

Comment:
I think you must add an else block that throws an exception. What do you think?

Reply:
This method is supposed to return an address to be used (to create change outputs, for example). I don't think we should raise an exception only because we've reached the gap limit, we should just return the current address (without generating a new one), until it receives a tx.

Reply:
It feels strange to stop generating new addresses. But I think we will review all these policies in the next weeks. So, for now, let's do this patch.